src/Symfony/Component/Translation/Translator.php

```diff
@@ -214,7 +214,9 @@ public function trans(?string $id, array $parameters = [], string $domain = null
             }
         }
 
-        if ($this->hasIntlFormatter && $catalogue->defines($id, $domain.MessageCatalogue::INTL_DOMAIN_SUFFIX)) {
+        if ($this->hasIntlFormatter
+            && ($catalogue->defines($id, $domain.MessageCatalogue::INTL_DOMAIN_SUFFIX)
+                || false != strstr($domain, MessageCatalogue::INTL_DOMAIN_SUFFIX))) {
```

Comment:
Instead of using `strstr()`, I'd suggest using `substr_compare()` with a negative offset